### PR TITLE
nspawn: Miscellaneous touch-up

### PR DESCRIPTION
**Pull request by LukeShu**
This PR doesn't really accomplish anything.  It consists of miscellaneous clean-ups to nspawn that are left over from other work I've done, and that I thought might be welcome upstream.  I won't be upset if any or all of the commits are rejected.

/cc @brauner I mentioned you in the `tmpfs_patch_options` patch.

### CODE REVIEW

**Comment by LukeShu** [comment]
v2: Use a consistent email for the author/committer information on the commits (oops).  No content change.

File: src/basic/cgroup-util.c [hunk 1]
```diff
@@ -2385,9 +2385,9 @@ int cg_kernel_controllers(Set **ret) {
         assert(ret);
 
         /* Determines the full list of kernel-known controllers. Might
-         * include controllers we don't actually support, arbitrary
-         * named hierarchies and controllers that aren't currently
-         * accessible (because not mounted). */
+         * include controllers we don't actually support and
+         * controllers that aren't currently accessible (because not
+         * mounted). */
```
Comment (poettering) [inline src/basic/cgroup-util.c]: this should really stay. I mean, the point of the comment is to say what the call returns, and not what the underlying kernel API does.

It's very interesting to know that cg_kernel_controllers() never returns named hierarchies, and that should be documented in a comment. Whether it is the function that filters them out or the kernel which doesn't even include them doesn't matter and is an implementation detail really.

File: src/nspawn/nspawn-cgroup.c [hunk 2]
```diff
@@ -5,14 +5,19 @@
...
-#include "nspawn-cgroup.h"
+#include "path-util.h"
 #include "rm-rf.h"
 #include "string-util.h"
 #include "strv.h"
+#include "user-util.h"
 #include "util.h"
 
+#include "nspawn-cgroup.h"
+#include "nspawn-mount.h"
```
Comment (poettering) [inline src/nspawn/nspawn-cgroup.c]: we don't separate these out usually. Only our public installed headers (i.e. those with an "sd-" prefix) are separated out

File: src/nspawn/nspawn-cgroup.c [hunk 3]
```diff
@@ -188,3 +193,418 @@ int create_subcgroup(pid_t pid, bool keep_unit, CGroupUnified unified_requested)
         (void) cg_enable_everywhere(supported, supported, cgroup);
         return 0;
 }
+
+/********************************************************************/
```
Comment (poettering) [inline src/nspawn/nspawn-cgroup.c]: huh, we usually don't decorate things with such lines

File: src/nspawn/nspawn-mount.c [hunk 4]
```diff
@@ -426,10 +426,17 @@ int mount_sysfs(const char *dest, MountSettingsMask mount_settings) {
...
+        /* We need to ensure that /sys/fs/cgroup exists before we remount /sys read-only.
+         *
+         * If !use_cgns, then this was already done by the outer child; so we only need to do it here it if use_cgns.
+         * This function doesn't know whether use_cgns, but !cg_ns_supported()⇒!use_cgns, so we can "optimize" the case
+         * where we _know_ !use_cgns, and deal with a no-op mkdir_p() in the false-positive where cgns_supported() but
+         * !use_cgns.
+         *
+         * But is it really much of an optimization?  We're potentially spending an access(2) (cg_ns_supported() could
+         * be cached from a previous call) to potentially save an lstat(2) and mkdir(2); and all of them are on virtual
+         * fileystems, so they should all be pretty cheap. */
```
Comment (poettering) [inline src/nspawn/nspawn-mount.c]: quite franky i'd just simplify this and drop the conditionalization, and just do the mkdir in all cases...

**Comment by poettering** [comment]
looks good, just a few comments

**Comment by LukeShu** [comment]
v3:
 - `cg_kernel_controllers()`: Add a sentence to the comment, clarifying that we're not including "name=" pseudo-controllers.
 - `nspawn-cgroup.c`: Don't separate `nspawn-*.h` out in the includes list.
 - `nspawn-cgroup.c`: Don't stick a `/*****/` divider between the non-mount and mount stuff.
 - `nspawn-mount.c`: Unconditionally mkdir /sys/fs/cgroup; move the formerly inserted comment explaining the conditional to the commit message, justifying its removal.

File: src/nspawn/nspawn-mount.c (hunk 4 above)
Comment (LukeShu) [reply src/nspawn/nspawn-mount.c]: /cc @brauner You're the author of that optimization.

**Pull request merged**